app/models/user.rb

```diff
@@ -15,7 +15,7 @@ class User < ApplicationRecord
   has_many :comments, dependent: :nullify
   belongs_to :local_authority, optional: false
 
-  before_create :generate_otp_secret
+  before_create :generate_otp_secret unless !Rails.env.development? || ENV["2FA_ENABLED"] == "true"
```

Comment:
That's really handy for our dev environment, can you please add it on a private method? There's the revert kind logic on [L11](https://github.com/unboxed/bops/blob/main/app/models/user.rb#L11)

Reply:
It only uses `database_authenticatable` in development and if 2FA has been disabled.
I think this is because the login mechanism expects `two_factor_authenticatable` instead.

You can try taking the line out and see if you can log in when 2FA is disabled. From memory it won't work